streaming/index.js

```diff
@@ -771,28 +755,15 @@ const startServer = async () => {
    * @param {any} req
    * @returns {Promise.<void>}
    */
-  const authorizeListAccess = (listId, req) => new Promise((resolve, reject) => {
+  const authorizeListAccess = async (listId, req) => {
     const { accountId } = req;
 
-    pgPool.connect((err, client, done) => {
-      if (err) {
-        reject();
-        return;
-      }
-
-      // @ts-ignore
-      client.query('SELECT id, account_id FROM lists WHERE id = $1 LIMIT 1', [listId], (err, result) => {
-        done();
-
-        if (err || result.rows.length === 0 || result.rows[0].account_id !== accountId) {
-          reject();
-          return;
-        }
+    const result = await pgPool.query('SELECT id, account_id FROM lists WHERE id = $1 AND account_id = $2 LIMIT 1', [listId, accountId]);
```

Comment:
The query does change here, but the performance should be the same. Arguably we could do this entirely on the index by having an index on `id, account_id`, but it's fast enough:

| QUERY PLAN |
| --- |
| Limit  \(cost=0.15..8.17 rows=1 width=16\) \(actual time=0.092..0.093 rows=1 loops=1\) |
|   \-\>  Index Scan using lists\_pkey on lists  \(cost=0.15..8.17 rows=1 width=16\) \(actual time=0.090..0.090 rows=1 loops=1\) |
|         Index Cond: \(id = 2\) |
|         Filter: \(account\_id = '112570031471696657'::bigint\) |
| Planning Time: 0.362 ms |
| Execution Time: 0.121 ms |